clustered/server/src/main/java/org/ehcache/clustered/lock/server/VoltronReadWriteLockActiveEntity.java

```diff
@@ -57,6 +60,25 @@ public LockTransition invoke(ClientDescriptor client, LockOperation message) {
   }
 
   @Override
+  public void addStateTo(StateDumpCollector dump) {
+    dump.addState("exclusiveHolder", String.valueOf(exclusiveHolder));
+    {
+      StateDumpCollector releaseListenersDump = dump.subStateDumpCollector("releaseListeners");
+      int idx = 0;
+      for (ClientDescriptor releaseListener : new HashSet<>(this.releaseListeners)) {
+        releaseListenersDump.addState(String.valueOf(idx++), releaseListener.toString());
+      }
+    }
+    {
+      StateDumpCollector sharedHolders = dump.subStateDumpCollector("sharedHolders");
```

Comment:
Shared holders should be dumped at the same level as exclusive holder. And actually it is an either or situation.